src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js

```diff
@@ -263,16 +265,16 @@ const filterCells = (cellSets, sampleKey, groupBy) => {
 const generateData = (cellSets, sampleKey, groupBy, embeddingData) => {
   const { filteredCells, cellSetLegendsData } = filterCells(cellSets, sampleKey, groupBy);
 
-  const plotData = filteredCells
-    .filter((d) => d.cellId < embeddingData.length)
-    .filter((data) => embeddingData[data.cellId]) // filter out cells removed in data processing
+  const plotData = embeddingData
+    .map((coordinates, cellId) => ({ cellId, coordinates }))
+    .filter(({ coordinates }) => coordinates !== undefined)
     .map((data) => {
-      const { cellId, ...supportingData } = data;
+      const { cellId, coordinates } = data;
 
       return {
-        ...supportingData,
-        x: embeddingData[cellId][0],
-        y: embeddingData[cellId][1],
+        ...filteredCells[cellId],
+        x: coordinates[0],
+        y: coordinates[1],
```

Comment:
Nice! way more readable.
Do you know what was the purpose of `.filter((d) => d.cellId < embeddingData.length)` by the way?

Reply:
I think it's just a short cut to truncate data that's longer than the embedding data returned from the worker.

Reply:
yes, although it is still weird, why would we need to perform the truncation